utils/pubsub.go

```diff
@@ -14,56 +14,108 @@ type DataEvent struct {
 // DataChannel is a channel which can accept an DataEvent
```

Comment:
Out of curiosity, why is this file in `utils` without a subpackage like the others (i.e. `utils/pubsub/pubsub.go`)?

Reply:
this is a good question!